Core/Src/state_machine.c

```diff
@@ -218,12 +190,23 @@ static int queue_state_transition(state_req_t new_state)
 		return 1;
 	}
 
-	return osMessageQueuePut(state_trans_queue, &new_state, 0U, 0U);
+	osStatus_t status =
+		osMessageQueuePut(state_trans_queue, &new_state, 0U, 0U);
```

Comment:
i dont understand why we need flags AND queues, feel like it should be one or the other, if it always has to have a flag, then we should make wrappers for putting to queues that automatically call both of these methods together and force us to always do that in the future